staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -297,6 +298,17 @@ func (flags *ApplyFlags) ToOptions(f cmdutil.Factory, cmd *cobra.Command, baseNa
 		return nil, err
 	}
 
+	var applySet *ApplySet
+	if flags.ApplySetRef != "" {
+		var applySetNs string
+		// ApplySet uses the namespace value from the flag, but not from the kubeconfig or defaults
```

Comment:
What is the reason of this?

Reply:
It is to be very explicit for everything related to the pruning operation, and to err on the side of being more restrictive to start and relaxing later if needed. Note that 'enforcing' the namespace on an object works the same way: if your object has no namespace in the config, one will be filled in based on the value of the namespace flag, but not the kubeconfig's current or default values.